narwhals/_expression_parsing.py

```diff
@@ -323,8 +342,9 @@ def with_elementwise_op(self) -> ExprMetadata:
         )
 
     def with_unorderable_window(self) -> ExprMetadata:
+        # Window function which cannot be used with `over(order_by=...)`.
         if self.is_scalar_like:
-            msg = "Can't apply unorderable window (`rank`, `is_unique`) to scalar-like expression."
+            msg = "Can't apply unorderable window (e.g. `is_unique`) to scalar-like expression."
             raise InvalidOperationError(msg)
         return ExprMetadata(
```

Comment:
i don't think it's worth having this complexity _just_ for a single expression (`is_unique`)

I think we can just bundle this together with `with_window`, and if anyone uses it with `order_by` then no big deal